drafts/bitcartel-nu0-network-handshake/draft.rst

```diff
@@ -0,0 +1,191 @@
+        ...
+    }
+      
+Where:
+
+- Pre-Overwinter PROTOCOL_VERSION is 170002
+- OVERWINTER_PROTO_VERSION is 170003
+
+
+Rejecting Pre-Overwinter Connections
+------------------------------------
+
+Currently, nodes will reject connections from nodes with a protocol version lower than the other node's minimum supported protocol version.
+
+This value is defined as::
+
+    //! disconnect from peers older than this proto version
+    static const int MIN_PEER_PROTO_VERSION = 170002;
+    
+With rejection codified as::
+    
+    if (pfrom->nVersion < MIN_PEER_PROTO_VERSION)
+    {
+        // disconnect from old peers running protocol version we don't support.
+
+Prior to activation, Overwinter nodes will contain the following constants::
+
+    static const int PROTOCOL_VERSION = 170003;
+    static const int MIN_PEER_PROTO_VERSION = 170002;
+
+This allows pre-Overwinter nodes (which only supports protocol version 170002) and Overwinter nodes (which support both 170002 and 170003) to remain connected prior to activation.
+
+However, once Overwinter activates, Overwinter nodes should:
+
+- reject new connections from pre-Overwinter nodes
+- disconnect any existing conncetions to pre-Overwinter nodes
+
+
+Network Coalescence
+-------------------
+
+Prior to the activation of Overwinter, nodes running pre-Overwinter protocol version 170002 and the Overwinter protocol version 170003 remain connected with the same consensus rules, but it would be preferable for nodes supporting Overwinter to connect to other nodes supporting Overwinter.
+
+This will help the network partition smoothly, since nodes should already be connected to (a majority of) peers running the same protocol version.  Otherwise am Overwinter node may find their connections to legacy nodes dropped at the activation height, leaving them isolated and potentially susceptible to eclipse attacks. [link]
+
+To assist network coalescence before the activation height, we update the eviction process to place a higher priority on evicting legacy nodes.
+
+This can be activated at n blocks before the activation block height, where n could be defined by a constant such as::
+
+    static const int NETWORK_COALESCE_BLOCK_PERIOD = 1000.
+
+The eviction code can be updated as follows::
+
+    static bool AttemptToEvictConnection(bool fPreferNewConnection) {
+    ...
+    // Protect connections with certain characteristics
+    ...
+    // Check version of eviction candidates
+    if (current_block_height >= (activationheight - NETWORK_COALESCE_BLOCK_PERIOD)) {
+      // if there exist any legacy nodes, keep them in the eviction set
+      // and at the same time remove overwinter nodes from eviction set.
+      // if there do not exist any legacy nodes,
+      // continue with existing behaviour.
+
+
+Disconnecting Existing Connections
+----------------------------------
+
+It is likely that at the activation block height, an Overwinter node will still be connected to some Legacy nodes.
```

Comment:
Why is this likely, if we started putting them in the eviction set 1000 blocks before? Maybe "possible" is better?

Reply:
This wasn't well worded, fixing.